### ISSUE TITLE
[sklearn#33914] `partial_fit` for `Ridge`

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#374]
**Mirrored from [scikit-learn#33914](https://github.com/scikit-learn/scikit-learn/issues/33914)**

Original title: `partial_fit` for `Ridge`
Original labels: New Feature
Created: 2026-05-01T08:24:52Z

---

### Describe the workflow you want to enable

The `Ridge` family is already compatible with the Array API, which is great for pytorch compatibility! It simplifies linear probing evaluations for foundation models. 

However, these models often use datasets with many examples.
It is not always feasible to accumulate features in memory before passing them to a ridge classifier.

It would be nice if `Ridge` had a `partial_fit` method which allowed processing the dataset in batches. 
With $D$ features and $N$ examples, `Ridge` would only accumulate the features statistics ($D \times D$) instead of all the features ($N \times D$)

### Describe your proposed solution

We accumulate the sufficient statistics $A = X^\top X \in \mathbb{R}^{D \times D}$ and $B = X^\top Y \in \mathbb{R}^{D \times C}$, both of size independent of $N$. The closed-form ridge solution is $W = (A + \lambda I)^{-1} B$. Because $X$ is never materialized, memory is $O(D^2)$ instead of $O(ND)$, so the probe scales to arbitrarily large train sets.

### Describe alternatives you've considered, if relevant

Implement it outside of scikit learn: https://github.com/braindecode/OpenEEGBench/blob/ede655f1294feef15ff6be4d6e6d94628ab9187e/open_eeg_bench/ridge_probe.py#L79-L273
Then add a test for consistency with the scikit learn pipeline on a small random datasets: https://github.com/braindecode/OpenEEGBench/blob/ede655f1294feef15ff6be4d6e6d94628ab9187e/tests/test_ridge_probe.py#L42-L99

### Additional context

We are using this for [OpenEEGBench](https://github.com/braindecode/OpenEEGBench), a library for benchmarking EEG foundation models. 

Context: https://github.com/braindecode/OpenEEGBench/pull/32#discussion_r3172514283

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to [24](https://github.com/godofecht/flow-scikit/issues/24) in the flow-scikit repository. Further library-specific triage and implementation will happen there.

**Comment by godofecht** [comment]
Moved to [25](https://github.com/godofecht/flow-scikit/issues/25) in the flow-scikit repository. Further library-specific triage and implementation will happen there.
